manager/scheduler/filter.go

```diff
@@ -217,3 +217,46 @@ func (f *ConstraintFilter) Explain(nodes int) string {
 	}
 	return fmt.Sprintf("scheduling constraints not satisfied on %d nodes", nodes)
 }
+
+// PlatformFilter selects only nodes that run the required platform.
+type PlatformFilter struct {
+	supportedPlatforms []*api.Platform
+}
+
+// SetTask returns true when the filter is enabled for a given task.
```

Comment:
the name `SetTask` makes this seem like a setter, with side effects, but the doc comment makes it seem like a comparison, without side effects. but the code body clearly shows side effects. can you rewrite this comment? is the boolean supposed to indicate the success or failure of the operation? if it fails, would it be better to return an `error`?